### ISSUE TITLE
Error message visuals

### ISSUE THREAD

**Issue by gnakm** [department-of-veterans-affairs/vets.gov-patterns#42]
The error messages are not visually consistent throughout Vets.gov. Considering there are many different types of errors and situations for them, we should be thoughtful about the placement for each one. We are also missing breadcrumbs on all the pages.

What we have right now:
<img width="1106" alt="screen shot 2016-12-06 at 3 31 43 pm" src="https://cloud.githubusercontent.com/assets/13420618/20990210/ac686a04-bca5-11e6-8393-33234a434505.png">

![screen_shot_2016-12-06_at_3 33 52_pm_720](https://cloud.githubusercontent.com/assets/13420618/20990228/bb586ac8-bca5-11e6-8e5b-e8e832426f75.png)

## Styles + Errors
1. **Incomplete action error**
- When an action didn't go through
- Examples: prescription couldn't be refilled, is not refillable, SM didn't go through.
- Visual: Red Alert UI with short bolded description and actionable step.
- Placement: Above main header because we jump to top of page for confirmations, looks more prominent and doesn't get lost within the page. 

<img width="374" alt="screen shot 2016-12-07 at 5 37 02 pm" src="https://cloud.githubusercontent.com/assets/13420618/20989781/f3b9d57a-bca3-11e6-9764-b3b67cf12ea9.png">

2. **Unavailable error**
- When user's information is unavailable/ downtime
- Examples: When the MHV is down, can't locate medications or messages, can't reach service
- Visual: Red Alert UI with short bolded description and actionable step.
- Placement: Under main header

<img width="390" alt="screen shot 2016-12-07 at 5 37 37 pm" src="https://cloud.githubusercontent.com/assets/13420618/20989792/fc5b6d42-bca3-11e6-8001-a6b998d4f893.png">

3. **Empty state**
- This is not an error; user doesn't have content. 
- Example: User is LOA3-ed but doesn't have MHV account or has MHV account but 0 active Rx (though we should add tabs to design because user might have expired Rx).
- Visual: Header, description and actionable step `.primary p`.

![rx - facility closed](https://cloud.githubusercontent.com/assets/13420618/20990463/e56dce74-bca6-11e6-8b9d-08e4a22370a4.png)

We went back and forth about using the yellow alert pattern but I think we shouldn't use it because with the MHV errors, it's hard to split them into two categories of error and the yellow one should be used for warning/ disclaimer purposes. 

Next steps:
- [ ] @emilyville and @melwoodard Could you guys review and approve of this iteration?
- [ ] @AnitraAppa Once ^^^ approve, could you revisit the [Vets.gov error messages](https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Design/Error-messaging.md) and make sure we have the latest errors? Some of the have xxx-xxx-xxx instead of actual number. I know there were many issues flying around for specific projects. Want to make sure master list is updated!
- [ ] Gina to go through all of the errors and indicate with type error it is with visual for reference. Tag design leads on each project to make an issue in respective repos and triage for implementation. 

go team!

**Comment by melwoodard** [comment]
Thanks for the thorough documentation @gnakm!

The distinctions between 1 and 2 make sense to me, and as an overall systems approach I think being flexible to adjust for context is more effective than setting rigid placement rules. 

For 3, I think your example gives two separate cases that should be treated differently. I'd say that not having an MHV account _is_ still an error (in that you may have VA prescriptions but don't have authorization to access them), while not having any VA prescriptions is a true empty state. As long as we're able to technically distinguish these two cases, we should treat them differently.

**Comment by AnitraAppa** [comment]
@gnakm Good call, will scrub through those again. At quick review though, it does look like I replaced all the number placeholders with the actual help line number. There are XXX'S where the API response field isn't filled in, but I don't see any missing numbers. Let me know if there's a particular one I'm missing! https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Design/Error-messaging.md
